runtime/src/lib.rs

```diff
@@ -443,6 +443,26 @@ impl_runtime_apis! {
 					.collect(),
 			}
 		}
+
+		/// Returns all the current auctions without auction id,
+		///  or return a specific auction info with an auction id.
+		fn current_auctions(auction_id: Option<AuctionId>) -> Vec<AuctionInfo>{
+			let data = pallet_auction::Auctions::<Runtime>::iter()
+			.map(|(auction, auction_info)|{
+				AuctionInfo { auction_id: auction, nft_id: auction_info.nft_id, start: auction_info.start, reserve: auction_info.reserve, buy_now: auction_info.buy_now, expiry_block: auction_info.expiry_block, current_bid: auction_info.current_bid }
+			}).collect();
+
+			match auction_id{
+				Some(id) => {
+					if let Some(auction_data) = pallet_auction::Auctions::<Runtime>::get(id){
```

Comment:
If auction ID is passed in, all the code above is wasted. You can have this match at the top and only execute the code above if no auction ID is passed in